code/modules/reagents/chemistry/reagents/food_reagents.dm

```diff
@@ -46,6 +47,23 @@
 		if (FOOD_AMAZING)
 			SEND_SIGNAL(exposed_mob, COMSIG_ADD_MOOD_EVENT, "quality_food", /datum/mood_event/amazingtaste)
 
+/datum/reagent/consumable/on_transfer(atom/A, methods, trans_volume)
+	. = ..()
+	to_chat(world.log, A.type)
+	var/obj/item/organ/stomach/belly = A
+	if(!istype(belly))
+		return .
+	if(nutriment_factor)
+		if(!HAS_TRAIT(belly.owner, TRAIT_NUTRIMENT_PASTE_FAN))
+			if((methods & EATEN_FOOD) != EATEN_FOOD)
```

Comment:
does this account for drinks as well? IIRC, most drinks are consumable subtypes.

Reply:
ath read the code and you'll see how it doesn't count against drinks